packages/accounts/src/msca/e2e-tests/multisig-modular-account.e2e.test.ts

```diff
@@ -0,0 +1,572 @@
+import {
+  LocalAccountSigner,
+  LogLevel,
+  Logger,
+  sepolia,
+  wrapSignatureWith6492,
+  parseFactoryAddressFromAccountInitCode,
+  type SmartAccountSigner,
+  type UserOperationFeeOptions,
+} from "@alchemy/aa-core";
+import {
+  http,
+  type Address,
+  type Chain,
+  type HDAccount,
+  createPublicClient,
+  fromHex,
+  pad,
+} from "viem";
+import { createMultisigModularAccountClient } from "../client.js";
+import { formatSignatures } from "../plugins/multisig/index.js";
+import { API_KEY, OWNER_MNEMONIC } from "./constants.js";
+
+const chain = sepolia;
+
+Logger.setLogLevel(LogLevel.DEBUG);
+
+describe("Multisig Modular Account Tests", async () => {
+  const signer1: SmartAccountSigner<HDAccount> =
+    LocalAccountSigner.mnemonicToAccountSigner(OWNER_MNEMONIC, {
+      accountIndex: 0,
+    });
+
+  const signer2: SmartAccountSigner<HDAccount> =
+    LocalAccountSigner.mnemonicToAccountSigner(OWNER_MNEMONIC, {
+      accountIndex: 1,
+    });
+
+  const signer3: SmartAccountSigner<HDAccount> =
+    LocalAccountSigner.mnemonicToAccountSigner(OWNER_MNEMONIC, {
+      accountIndex: 2,
+    });
+
+  const threshold = 2n;
+
+  const owners = [
+    await signer1.getAddress(),
+    await signer2.getAddress(),
+    await signer3.getAddress(),
+  ];
+
+  it("should successfully get counterfactual address", async () => {
+    const {
+      account: { address },
+    } = await givenConnectedProvider({
+      signer: signer1,
+      chain,
+      owners,
+      threshold,
+    });
+    expect(address).toMatchInlineSnapshot(
+      '"0x4ff93F25764CefC22aeeE111CEf47CD1B5e05370"'
+    );
+  });
+
+  it("should get on-chain owner address successfully", async () => {
+    const provider = await givenConnectedProvider({
+      signer: signer1,
+      chain,
+      owners,
+      threshold,
+    });
+
+    // .map(...) just to generate a non-readonly copy
+    expect((await provider.readOwners()).slice().sort()).toMatchInlineSnapshot(
```

Comment:
I would use `toBe` instead of inline snapshot

Reply:
It looks like `toBe` didn't work with the array, so I used `toStrictEqual`. `toBe` gave me this error:
```
AssertionError: expected [ …(3) ] to be [ …(3) ] // Object.is equality

If it should pass with deep equality, replace "toBe" with "toStrictEqual"

Expected: [ …(3) ]
Received: serializes to the same string
```